src/Set.js

```diff
@@ -82,7 +82,19 @@ export class Set extends SetCollection {
   // @pragma Composition
 
   map(mapper, context) {
-    return updateSet(this, this._map.map(v => mapper(v, v, this), context));
+    return this.withMutations(set => {
+      const removes = [];
+      const adds = [];
+      this.forEach(value => {
+        const mapped = mapper.call(context, value, value, set);
```

Comment:
The last value here should be `this` - the original immutable map instead of the transiently mutable one.